gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/version/HiveDatasetVersionCleaner.java

```diff
@@ -87,21 +87,25 @@ public void clean() throws IOException {
     try (AutoReturnableObject<IMetaStoreClient> client = cleanableHiveDataset.getClientPool().getClient()) {
       Partition partition = hiveDatasetVersion.getPartition();
       try {
+        if (cleanableHiveDataset.isShouldDeleteData()) {
+          cleanableHiveDataset.getFsCleanableHelper().clean(hiveDatasetVersion, possiblyEmptyDirectories);
+        }
```

Comment:
If we are doing this first before dropping the partitions, then we should also check if the flow is a simulate flow. Otherwise we will delete the underlying data before it simulates dropping the partition. Also I think this was a bug in the existing code as well, before it would also delete the underlying data if it was simulate.

Reply:
Yeah, I thought that was the expected flow since we used to allow deleting of the underlying files in simulation mode. Good that we identified that bug too :)! I have pushed my changes in the latest PR